src/usdb_syncer/gui/mw.py

```diff
@@ -281,6 +311,21 @@ def _open_current_song_folder(self) -> None:
         else:
             _logger.info("No current song.")
 
+    def _open_current_song_in_app(self, app: settings.SupportedApps) -> None:
+        if song := self.table.current_song():
+            if song.sync_meta:
+                if song.sync_meta.path.exists():
+                    settings.SupportedApps.open_app(app, song.sync_meta.path.parent)
+                else:
+                    with db.transaction():
+                        song.remove_sync_meta()
+                    events.SongChanged(song.song_id)
+                    _logger.info("Song does not exist locally anymore.")
+            else:
+                _logger.info("Song does not exist locally.")
+        else:
+            _logger.info("No current song.")
+
```

Comment:
This function is an almost identical duplicate. The common parts should be refactored into a separate function.